R/fit.R

```diff
@@ -117,45 +109,45 @@ h_summarize_all_fits <- function(all_fits) {
 #' )
 #' best_fit <- refit_multiple_optimizers(fit)
 refit_multiple_optimizers <- function(fit,
-                                      n_cores = 1L,
-                                      optimizers = c("L-BFGS-B", "BFGS", "CG", "nlminb"),
-                                      accept_singular = TRUE) {
+                                      ...,
+                                      control = mmrm_control(...)) {
   assert_class(fit, "mmrm_fit")
-  assert_int(n_cores, lower = 1L)
-  optimizers <- match.arg(optimizers, several.ok = TRUE)
-
+  assert_class(control, "mmrm_control")
   # Extract the components of the original fit.
   old_formula <- formula(fit)
   old_data <- fit$data
   old_weights <- fit$weights
-  old_optimizer <- attr(fit, "optimizer")
```

Comment:
because of the fact that the label of the optimizer can be really custom, and it can be hard to infer the optimizer used, e.g. fit_single_optimizer using  "BFGS" different start/control can lead to fail, and if we change the control use refit_multiple_optimizers, it is strange that we omit the optimizer we used (as they are using different control/start or other args). So I moved the logic of "not use the already fitted" into `mmrm`, skipping the first optimizer in `mmrm`.